nixos/modules/services/backup/tsm.nix

```diff
@@ -0,0 +1,159 @@
+      '';
+      description = ''
+        <literal>include.*</literal> and
+        <literal>exclude.*</literal>
+        directives for system backup.
+        The default value excludes compressed
+        formats from being recompressed by TSM.
+        To disable the default value,
+        you have to use <function>mkForce</function>.
+      '';
+    };
+    extraConfig = lib.options.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        encryptiontype  AES256
+        encryptkey  save
+      '';
+      description = ''
+        Additional system-options file (dsm.sys) directives.
+      '';
+    };
+    autoTime = lib.options.mkOption {
+      type = lib.types.nullOr (lib.types.strMatching ".+");
+      default = null;
+      example = "12:00";
+      description = ''
+        Backup will be invoked automatically
+        at the given date/time,
+        which must be in the format described in
+        <citerefentry><refentrytitle>systemd.time</refentrytitle><manvolnum>5</manvolnum></citerefentry>.
+        <option>generatePwd</option> must be enabled for this!
+        The default <literal>null</literal>
+        disables automatic backups.
+      '';
+    };
+  };
+
+  cfg = config.services.tsmBackup;
+
+  assertion = {
+    assertion = (cfg.autoTime!=null)->cfg.generatePwd;
+    message = "TSM auto backup requires automatic password generation";
+  };
+
+  # Client system-options file directives are explained here:
+  # https://www.ibm.com/support/knowledgecenter/SSEQVQ_8.1.7/client/c_opt_usingopts.html
+  dsmSysText = ''
+    ****  IBM Spectrum Protect (Tivoli Storage Manager)
+    ****  client system-options file "dsm.sys".
+    ****  Do not edit!
+    ****  This file is generated by NixOS configuration.
+    servername  systembackup
+      commmethod  ${if cfg.ip6 then "v6tcpip" else "tcpip"}
+      tcpserveraddress  ${cfg.server}
+      tcpport  ${builtins.toString cfg.port}
+      nodename  ${cfg.node}
+      passwordaccess  ${if cfg.generatePwd then "generate" else "prompt"}
+      inclexcl  ${pkgs.writeText "dsm.sys-inclexcl" cfg.includeExclude}
+    compression  yes
+    passworddir  /var/lib/${etcDir}/password/
+    errorlogname  /var/log/tsm-client.log
+    ****  extra config
+    ${cfg.extraConfig}
+  '';
+
+  service = lib.modules.mkIf (cfg.autoTime!=null) rec {
+    description = "IBM Spectrum Protect (Tivoli Storage Manager) System Auto Backup";
+    environment.DSM_CONFIG = "/dev/null";
+    # TSM needs a HOME dir to store certificates.
+    environment.HOME = "/var/lib/${serviceConfig.StateDirectory}/home";
+    serviceConfig.ExecStartPre = "${pkgs.coreutils}/bin/mkdir --parents $HOME";
+    # for exit status description see
+    # https://www.ibm.com/support/knowledgecenter/en/SSEQVQ_8.1.7/client/c_sched_rtncode.html
+    serviceConfig.SuccessExitStatus = "4 8";
+    serviceConfig.ExecStart = "${tsmPkg}/bin/dsmc backup";
+    serviceConfig.StateDirectory = etcDir;
```

Comment:
This will create `/var/lib/tsm-client`.

Can you explain a bit what lands where, and why `HOME` is set like above?